src/main/java/org/jabref/gui/maintable/columns/FileColumn.java

```diff
@@ -98,8 +98,35 @@ public FileColumn(MainTableColumnModel model,
                                          .getGraphicNode());
 
         new ValueTableCellFactory<BibEntryTableViewModel, List<LinkedFile>>()
-                .withGraphic((entry, linkedFiles) -> createFileIcon(entry, linkedFiles.stream().filter(linkedFile ->
-                                linkedFile.getFileType().equalsIgnoreCase(fileType)).collect(Collectors.toList())))
+                .withGraphic((entry, linkedFiles) -> createFileIcon(entry, linkedFiles.stream()
+                                                                                      .filter(linkedFile -> linkedFile.getFileType().equalsIgnoreCase(fileType))
+                                                                                      .collect(Collectors.toList())))
+                .withOnMouseClickedEvent((entry, linkedFiles) -> event -> {
+                    List<LinkedFile> filteredFiles = linkedFiles.stream()
+                                                                .filter(linkedFile -> linkedFile.getFileType().equalsIgnoreCase(fileType))
+                                                                .collect(Collectors.toList());
+
+                    if (event.getButton() == MouseButton.PRIMARY) {
+                        if (filteredFiles.size() == 1) {
+                            // Only one file - open directly
+                            LinkedFileViewModel linkedFileViewModel = new LinkedFileViewModel(filteredFiles.get(0),
+                                    entry.getEntry(), database, taskExecutor, dialogService, preferences);
+                            linkedFileViewModel.open();
+                        } else if (filteredFiles.size() > 1) {
+                            // Multiple files - show context menu to choose file
+                            ContextMenu contextMenu = new ContextMenu();
+                            for (LinkedFile linkedFile : filteredFiles) {
+                                LinkedFileViewModel linkedFileViewModel = new LinkedFileViewModel(linkedFile,
+                                        entry.getEntry(), database, taskExecutor, dialogService, preferences);
+                                MenuItem menuItem = new MenuItem(linkedFileViewModel.getTruncatedDescriptionAndLink(),
+                                        linkedFileViewModel.getTypeIcon().getGraphicNode());
+                                menuItem.setOnAction(e -> linkedFileViewModel.open());
+                                contextMenu.getItems().add(menuItem);
+                            }
+                            contextMenu.show(((Node) event.getSource()), event.getScreenX(), event.getScreenY());
```

Comment:
This seems slightly weird to me.
@calixtus do we need casts like that?

Reply:
Maybe. Could be possible that event.getSource only returns an Object. Is there a typecheck somewhere before the cast?
Isn't there maybe some default behavior to open context menus to open at the mouse position instead of explicitly defining the coordinates?

Reply:
we do the same here https://github.com/JabRef/jabref/blob/db9f83cf77b1a5215e729d3c336371b4edef04a4/src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java#L48-L55

Reply:
Well, then I guess it's good.